### PR TITLE
[MOB-14663] Remove AID generation logic.

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPAnalytics/Sources/Analytics.swift [hunk 1]
```diff
@@ -310,86 +313,13 @@ public class Analytics: NSObject, Extension {
...
-    private func retrieveAnalyticsId(event: Event) {
-        // check if analytics state contains an RSID and host OR if privacy opt-out. if so, update shared state with empty id.
-        if !analyticsState.isAnalyticsConfigured() || analyticsState.privacyStatus == .optedOut {
-            Log.debug(label: LOG_TAG, "sendAnalyticsIdRequest - Analytics is not configured or privacy is opted out, the analytics identifier request will not be sent.")
-            analyticsProperties.setAnalyticsIdentifier(aid: nil)
-            analyticsProperties.setVisitorIdentifier(vid: nil)
-            // create nil shared state  and dispatch this data in a response event for any extensions listening for AID/VID change
-            dispatchAnalyticsIdentityResponse(event: event)
-            return
-        }
```
Comment (kevinlind) [inline AEPAnalytics/Sources/Analytics.swift:344]: In C++ and Android Analytics we kept this section. However, I'm okay with removing it as it's an odd side-effect of calling this method. Let's add a task to remove this from C++ and Android as well so the behavior is the same.

File: AEPAnalytics/Sources/Analytics.swift [hunk 2]
```diff
@@ -276,14 +276,17 @@ public class Analytics: NSObject, Extension {
...
-        if let eventData = event.data, !eventData.isEmpty {
-            if let vid = eventData[AnalyticsConstants.EventDataKeys.VISITOR_IDENTIFIER] as? String, !vid.isEmpty {
-                // set VID request
-                updateVisitorIdentifier(event: event, vid: vid)
-            }
-        } else { // get AID/VID request
-            retrieveAnalyticsId(event: event)
+        if analyticsState.privacyStatus == .optedOut {
+            Log.debug(label: LOG_TAG, "handleAnalyticsRequestIdentityEvent - Privacy is opted out, ignoring the update visitor identifier request.")
+            return
```
Comment (kevinlind) [inline AEPAnalytics/Sources/Analytics.swift:281]: It looks like this method handles two cases; 1) set VID and 2) get VID/AID. With this change, if get AID/VID is called and privacy is opted out, then no response is returned. Can you change this method to perform this check only when setting the VID?

**Review by kevinlind — CHANGES_REQUESTED** [review]
Unless I'm not seeing these tests, can you add tests to verify `handleAnalyticsRequestIdentityEvent` will dispatch 
1) AID = nil if no AID is persisted
2) AID = value if AID is persisted

File: AEPAnalytics/Sources/Analytics.swift (hunk 1 above)
Comment (praveek) [reply AEPAnalytics/Sources/Analytics.swift:344]: We already clear the IDs when privacy status changes to `optedOut`. This logic is redundant. I will add a task to remove this logic from C++ and Android.

**Comment by praveek** [comment]
Some of the tests are already present in AnalyticsIdTests. I have added more tests to cover other cases.

**Review by kevinlind — APPROVED** [review]

**Pull request merged**